builtin/logical/database/path_roles.go

```diff
@@ -194,7 +209,12 @@ func (b *databaseBackend) pathRoleCreateUpdate() framework.OperationFunc {
 			}
 
 			if revocationStmtsRaw, ok := data.GetOk("revocation_statements"); ok {
-				role.Statements.Revocation = revocationStmtsRaw.([]string)
+				revStmts := revocationStmtsRaw.([]string)
+				revStmts = strutil.RemoveEmpty(revStmts)
+				// Only overwrite if they provided real data
+				if len(revStmts) > 0 {
```

Comment:
If I want to remove all revocation statements, can I still do so with this check in place?

Reply:
Good point, will remove.